### PR TITLE
Feat/thread support

### PR DESCRIPTION
**Pull request by supercrafter100**

### CODE REVIEW

File: config.example.json [hunk 1]
```diff
@@ -14,5 +14,41 @@
...
+        "> Click the button at the bottom of this embed to start the process of creating a thread.",
+        "",
+        "**Some timps to get a quick answer:**",
+        "> • Make sure your issue has a **meaningful title**",
+        ">  ✅ \"*PHPMailer isn't sending emails*\", \"*Error during installation*\", \"*How do I get custom badge colours*\"",
+        ">  ❌ \"Help\", \"My website doesn't work\", \"Hello\"",
+        "> ** **",
+        "> • Provide **all the information** we need to look into the problem",
+        "> - The exact error you see (do not rephrase the error message!) (*including the full stack trace*)",
+        "> - The code you suspect to be the issue and a bit of context",
```
Comment (Aberdeener) [inline config.example.json:40]: We cannot expect 99% of our users to find the code which is causing the issue...

File: config.example.json [hunk 2]
```diff
@@ -14,5 +14,41 @@
...
+    "supportEmbedDescription": [
+        "**Things we do not help with:**",
+        "> • If you are experiencing issues with third party templates or modules, please ask the template authors for support. We cannot assist you with them.",
+        "> • When you are using a pirated template or module. (this will get you banned)",
+        "> • If you are using a very old version of NamelessMC. Newer versions came out with a reason so don't expect support for older versions.",
+        "",
+        "**Before you create a thread:**",
+        "> • `CTRL` + `F` and see if the question has already been answered",
+        "> • See if you can find the information on the documentation (https://docs.namelessmc.com/)",
+        "> • Known issues and workarounds: https://github.com/NamelessMC/Nameless/issues/2822",
```
Comment (Aberdeener) [inline config.example.json:28]: Can we change this to "Current known issues and workarounds: ..." ?

File: config.example.json [hunk 3]
```diff
@@ -14,5 +14,41 @@
...
     ],
     "welcomeChannelId": "335964434331860992",
-    "botowners": ["314046311035043840"]
+    "botowners": ["314046311035043840"],
+    "supportChannelId": "247865182477484032",
+    "supportEmbedDescription": [
+        "**Things we do not help with:**",
+        "> • If you are experiencing issues with third party templates or modules, please ask the template authors for support. We cannot assist you with them.",
+        "> • When you are using a pirated template or module. (this will get you banned)",
+        "> • If you are using a very old version of NamelessMC. Newer versions came out with a reason so don't expect support for older versions.",
```
Comment (Aberdeener) [inline config.example.json:23]: > Newer versions came out with a reason so don't expect support for older versions.
This sounds overly snarky and doesn't really convey any information.

"If you are using an old version of NamelessMC. It is difficult to provide support for old versions, as the issue may have been solved in a more recent update." maybe?

File: config.example.json [hunk 4]
```diff
@@ -14,5 +14,41 @@
...
+        "> • Make sure your issue has a **meaningful title**",
+        ">  ✅ \"*PHPMailer isn't sending emails*\", \"*Error during installation*\", \"*How do I get custom badge colours*\"",
+        ">  ❌ \"Help\", \"My website doesn't work\", \"Hello\"",
+        "> ** **",
+        "> • Provide **all the information** we need to look into the problem",
+        "> - The exact error you see (do not rephrase the error message!) (*including the full stack trace*)",
+        "> - The code you suspect to be the issue and a bit of context",
+        "> - What you did when the error happened",
+        "> ** **",
+        "> • More elaborate resources on smart questions",
```
Comment (Aberdeener) [inline config.example.json:43]: Both of these links are probably out of scope for most NamelessMC support questions, I don't think we need to have them here

File: src/events/supportInteractionCreate.event.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+                const titleInput = new TextInputComponent();
+                titleInput.setCustomId("support-title");
+                titleInput.setLabel("Issue");
+                titleInput.setStyle("SHORT");
+
+                const descriptionInput = new TextInputComponent();
+                descriptionInput.setCustomId("support-description");
+                descriptionInput.setLabel("Description");
+                descriptionInput.setStyle("PARAGRAPH");
```
Comment (Aberdeener) [inline src/events/supportInteractionCreate.event.ts:30]: Just wondering, does this allow them to drop images in? Or is it a text only area?

File: src/events/supportInteractionCreate.event.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+} from "discord.js";
+import { Event } from "../handlers/EventHandler";
+
+export default class InteractionCreate extends Event<"interactionCreate"> {
+    public event = "interactionCreate";
+
+    public async run(interaction: Interaction) {
+        if (interaction.isButton()) {
+            const buttonId = interaction.customId;
+            if (buttonId == "get-support") {
```
Comment (Aberdeener) [inline src/events/supportInteractionCreate.event.ts:17]: `===` here?

**Review by Aberdeener — CHANGES_REQUESTED** [review]

File: src/events/supportInteractionCreate.event.ts (hunk 5 above)
Comment (supercrafter100) [reply src/events/supportInteractionCreate.event.ts:30]: Discord its modal feature does not support adding anything else than text fields yet. Maybe in the future

**Pull request merged**